packages/evm/test/safeguards/IntentsValidator.test.ts

```diff
@@ -37,422 +40,582 @@ describe('IntentsValidator', () => {
     validator = await ethers.deployContract('IntentsValidator')
   })
 
-  describe('Swap modes', () => {
-    context('None', () => {
-      const intent = createSwapIntent()
-      const safeguard = createSafeguardNone()
+  describe('List', () => {
+    describe('Swap modes', () => {
+      context('None', () => {
+        const intent = createSwapIntent()
+        const safeguard = createSafeguardNone()
 
-      it('always reverts with IntentsValidatorNoneAllowed', async () => {
-        await expect(validator.validate(intent, [safeguard])).to.be.revertedWithCustomError(
-          validator,
-          'IntentsValidatorNoneAllowed'
-        )
+        it('always reverts with IntentsValidatorNoneAllowed', async () => {
+          await expect(validator.validate(intent, createListSafeguard(safeguard))).to.be.revertedWithCustomError(
+            validator,
+            'IntentsValidatorNoneAllowed'
+          )
+        })
       })
-    })
 
-    context('SourceChain', () => {
-      const intent = createSwapIntent({ sourceChain: CHAIN_LOCAL, destinationChain: CHAIN_LOCAL })
+      context('SourceChain', () => {
+        const intent = createSwapIntent({ sourceChain: CHAIN_LOCAL, destinationChain: CHAIN_LOCAL })
 
-      context('when the source chain is not denied', () => {
-        const safeguard = createOnlyChainSafeguard(SwapSafeguardMode.SourceChain, CHAIN_LOCAL)
+        context('when the source chain is not denied', () => {
+          const safeguard = createOnlyChainSafeguard(SwapSafeguardMode.SourceChain, CHAIN_LOCAL)
 
-        it('passes', async () => {
-          await expect(validator.validate(intent, [safeguard])).to.not.be.reverted
+          it('passes', async () => {
+            await expect(validator.validate(intent, createListSafeguard(safeguard))).to.not.be.reverted
```

Comment:
All `passes` cases need to be changed.
```suggestion
           expect(await validator.validate(intent, createListSafeguard(safeguard))).to.not.be.reverted
```